client/v3/client.go

```diff
@@ -502,15 +502,15 @@ func (c *Client) checkVersion() (err error) {
 					return
 				}
 			}
-			if maj < 3 || (maj == 3 && min < 2) {
+			if maj < 3 || (maj == 3 && min < 4) {
```

Comment:
We should avoid hard code the version in source code. For stable releases (e.g. 3.5 and 3.4), it's fine. For the main branch, let's raise a ticket to programmatically get the supported versions.

Note we only support two releases, which are 3.5 and 3.4 for now. FYI. https://etcd.io/docs/v3.5/op-guide/versioning/

Reply:
Can I interpret is as we support the current version and one previous version only? 
i.e. for v3.6, v3.5 is valid, and for v3.7, v3.6 is still valid, while 3.5 will be out of date.

Reply:
> Can I interpret is as we support the current version and one previous version only?

YES. Currently we support 3.4 and 3.5. When 3.6.0 is formally released, then 3.4 will be out of support. When 3.7.0 is released, then 3.5 will be out of support.

Reply:
Raised #16273

Reply:
thx